src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy

```diff
@@ -214,9 +214,9 @@ public class ModelInterpreter implements Serializable {
     def dockerOrWithout(Agent agent, Closure body) {
         if (agent.docker != null) {
             return {
-                script.getProperty("docker").image(agent.docker).inside {
+                script.getProperty("docker").image(agent.docker).inside(agent.args, {
```

Comment:
It works, the default parameter is an empty string but I didn't dig any deeper. I added test cases to catch if anything changes on the side of the docker plugin.